js/modules/enable-character-count.js

```diff
@@ -117,61 +109,64 @@ const enableCharacterCount = new function() {
     }
   }
 
-  this.onKeyUp = (e) => {
-    const { target, key } = e;
-    const { dataset } = target;
-
-    if (dataset.hasCharacterCount) {
-      const inputLength = target.value.length;
-      const { maxLength } = target;
-
-      timeout && clearTimeout(timeout);
-
-
-    
-      writeCharCount(target);
-      liveRegion.innerHTML = '';
-
-      if (inputLength > maxLength - globalWarningThreshold && !wasArrowPressed(key)) {
-        timeout = setTimeout(() => {
-          announceCharCount(target);
-        }, 1000);
-      }
-
-
+  function onKeyUp(event) {
+    const { target, key } = event;
+    const { dataset, maxLength } = target;
+    writeCharCount(target);
+    const inputLength = target.value.length;
+    const warningThreshold = dataset.warningThreshold ?? 20;
+    const isWithinWarningThreshold = inputLength > (maxLength - warningThreshold);
+    if (isWithinWarningThreshold && !wasArrowPressed(key)) {
+      announceCharacterCountWithDelay(target, 1000);
     }
   }
 
-  this.onFocus = (e) => {
-    const { target } = e;
+  function onKeyDown(event) {
+    const { target, key } = event;
     const { dataset } = target;
+    if (isReadCharacterCountKeyPressed(key, dataset.readCountKey)) {
+      announceCharacterCount(target);
+    }
+  }
 
-    if (dataset.hasCharacterCount) {
-      announceCharCount(target);
+  function isReadCharacterCountKeyPressed(keyPressed, readCountKey) {
+    if (readCountKey) {
+      return keyPressed === readCountKey;
+    } else {
+      return keyPressed === defaultReadCharacterCountKey;
     }
```

Comment:
(Optional) Technically, this can be slightly simplified to the following:

```suggestion
    if (readCountKey) {
      return keyPressed === readCountKey;
    }
    return keyPressed === defaultReadCharacterCountKey;
```